scripts/irods/upgrade_configuration.py

```diff
@@ -92,6 +92,36 @@ def upgrade(irods_config):
                                 'Please resolve this naming conflict manually and try again.' % (old_path, new_path, new_path))
                     shutil.move(old_path, new_path)
 
+    if old_version_tuple < (4, 3, 0):
+        # Load server_config.json.
+        with open(paths.server_config_path()) as f:
+            server_config = json.load(f)
+
+        # Set default values for the hosts_config.json and host_access_control_config.json.
+        server_config['host_access_control'] = {'access_entries': []}
+        server_config['host_resolution'] = {'host_entries': []}
+
+        # Merge hosts_config.json into server_config.json if it is available.
+        if os.path.exists(paths.hosts_config_path()):
+            with open(paths.hosts_config_path()) as f:
+                hosts_config = json.load(f)
+
+            if 'host_entries' in hosts_config and len(hosts_config['host_entries']) > 0:
+                # Ignores the schema information in the JSON file.
+                server_config['host_resolution']['host_entries'] = hosts_config['host_entries']
+
+        # Merge host_access_control_config.json into server_config.json if it is available.
+        if os.path.exists(paths.host_access_control_config_path()):
+            with open(paths.host_access_control_config_path()) as f:
+                host_access_control = json.load(f)
+
+            if 'access_entries' in host_access_control and len(host_access_control['access_entries']) > 0:
+                # Ignores the schema information in the JSON file.
+                server_config['host_access_control']['access_entries'] = host_access_control['access_entries']
+
+        # Write changes to disk.
+        irods_config.commit(server_config, paths.server_config_path(), make_backup=True)
```

Comment:
@trel @alanking Any thoughts on this upgrade code for hosts_config.json and host_access_control_config.json?

Reply:
We need to consider what happens if someone has defined `host_resolution` and/or `host_access_control` in their server_config.json prior to an upgrade from 4.2.

Reply:
Done.